addons/l10n_it_edi_withholding/models/account_move.py

```diff
@@ -252,3 +239,47 @@ def _l10n_it_edi_import_line(self, element, move_line_form, extra_info=None):
                         move_line_form.tax_ids |= pension_fund_tax
 
         return messages_to_log
+
+    def _l10n_it_edi_import_invoice(self, invoice, data, is_new):
+        """ Handle the case where ENASARCO pension fund contribution should be applied on the invoice globally.
+        In this case, there should only be one element with ENASARCO and these conditions should be fulfilled:
+         - AliquotaIVA is defined
+         - PrezzoUnitario == 0.0
+         - a corresponding DatiRiepilogo with the same AliquotaIVA and a ImponibileImporto
+        """
+        res = super()._l10n_it_edi_import_invoice(invoice=invoice, data=data, is_new=is_new)
+        if not res:
+            return
+        self = res
+        tree = data['xml_tree']
+        global_enasarco_lines = []
+        for additional_data_element in tree.xpath('//AltriDatiGestionali'):
+            data_kind = additional_data_element.xpath('./TipoDato')[0].text.lower()
+            data_text = additional_data_element.xpath('./RiferimentoTesto')[0].text.lower()
+            if data_kind == 'cassa-prev' and ('enasarco' in data_text or 'tc07' in data_text):
+                parent_element = additional_data_element.xpath('..')[0]
+                price_unit = get_float(parent_element, './PrezzoUnitario')
+                if price_unit == 0.0:
+                    global_enasarco_lines.append(parent_element)
+
+        if len(global_enasarco_lines) == 1:
+            parent_element = global_enasarco_lines[0]
+            enasarco_amount = get_float(parent_element, './AltriDatiGestionali/RiferimentoNumero')
+            price_unit = get_float(parent_element, './PrezzoUnitario')
+            vat_rate = get_float(parent_element, './/AliquotaIVA')
+            base_amount = self._get_l10_it_edi_get_taxable_amount_from_summary_data(parent_element.xpath('..')[0], vat_rate)
+            enasarco_percentage = -self.currency_id.round(enasarco_amount / base_amount * 100) if base_amount else 0.0
```

Comment:
Sorry, I'm realizing just now.

I see you are taking the Imponibile (base_amount) and then computing the ENASARCO percentage doing (enasarco_amount / base_amount for 22% VAT * 100).
This works if the ENASARCO is related to single lines with taxes with the same rate.

If we got a "global" line though, with price == 0, I think it's referred to the whole invoice, not only to the lines with 22% VAT. I think we should use the sum of all base amounts.

As I told you, this is not a "standard" way to use the XML schema. 🤷🏻 
It's a workaround for some vendors that do not know how to write the XML schema correctly.

Reply:
Not sure about this, but I can use `amount_untaxed` of the move instead of fetching `ImponibileImporto`.
It will simplify the fix.

But then, it means that `AliquotaIVA` of the parent of the ENASARCO element is useless and potentially wrong if there are taxes with different rates. :thinking:

Reply:
It is useless, and potentially wrong. Yes.
The whole "zero priced line" makes no sense. It shouldn't even be an invoice line, but vendors don't care and write XML that way nevertheless.

Don't use `amount_untaxed`, as later versions will not set it on the move until we exit the function. See the old PR's forward-port runbot issues.
![image](https://github.com/user-attachments/assets/c7c8e260-c3a7-4583-8651-d5d2e093307e)

Just sum all the values in the XML nodes for the Imponibile.

Reply:
I've updated the fix. :wink: